src/test/java/io/appium/java_client/proxy/ProxyHelpersTest.java

```diff
@@ -133,4 +164,54 @@ public Object onError(Object obj, Method method, Object[] args, Throwable e) thr
                         "onError get")
         )));
     }
+
+
+    @Test
+    void shouldFireEventsForRemoteWebElement() throws MalformedURLException {
+        final StringBuilder acc = new StringBuilder();
+        MethodCallListener listener = new MethodCallListener() {
```

Comment:
I'm not sure we need the below expression as we can replace `new MethodCallListener` with `new RemoteWebElementListener` right here and a proper instance is going to be created.

Also not sure about the class name as it not only listens to webelement-specific methods but also to webdriver ones